src/Guide.php

```diff
@@ -2,23 +2,44 @@
 
 namespace Vinelab\Country;
 
+use Illuminate\Contracts\Config\Repository;
+use Illuminate\Support\Arr;
+
 class Guide
 {
-    public function __construct(\Illuminate\Config\Repository $config)
+    /**
+     * @var Repository
+     */
+    private $config;
+
+    /**
+     * Guide constructor.
+     * @param Repository $config
+     */
+    public function __construct(Repository $config)
     {
         $this->config = $config;
     }
 
-    public function name($abbreviation)
+    /**
+     * @param string $abbreviation
+     * @return string|array
+     */
+    public function name(string $abbreviation)
     {
         $abbreviation = strtoupper($abbreviation);
 
-        return $this->config->get("countries.{$abbreviation}");
+        return Arr::get($this->config->get('countries'), $abbreviation);
     }
 
-    public function abbreviation($name)
+    /**
+     * @param string $name
+     * @return string|null
+     */
+    public function abbreviation(string $name)
     {
         $countries = $this->config->get('countries');
+
         if (array_search(ucwords($name), $countries)) {
```

Comment:
would this be better assigned to a variable and then returned? e.g.

```php
$found = array_search(ucwords($name), $countries);

if ($found) {
  return $found
} else {
  ...
}

```

Reply:
@Mulkave Sure, I took it a bit further and also extracted the logic into `scanStringNames` and `scanArrayNames` methods, because it was hard to understand what's going when looking at this condition.